tests/test_main.py

```diff
@@ -0,0 +1,64 @@
+"""
+Test for main page using fastapi test client.
+"""
+from datetime import datetime
+from fastapi.testclient import TestClient
```

Comment:
What is the advantage of using fastapi.testclient over requests? I felt like it is offering very little here in terms of functionality. If anything, it will make it hard to use with our framework.

BTW, I am ok with this test alone using this client since we want to try stuff out like this all the time.

Reply:
Frankly even I don't get any close to specific advantage for fastapi.testclient as it uses requests module as wrapper . I choose it because we have developed the code using FastAPI. One use case I found at this stack article, is that user is using requests module to test API endpoints. However he also tried testclient as he thought it will help him to setup the test in the function of form. I think we can try this with other frameworks such as Django or even some other application involving API's and may be we can get more idea on which to choose. I believe test client will have all capabilities of request module.

https://stackoverflow.com/questions/60783222/how-to-test-a-fastapi-api-endpoint-that-consumes-images